rosbag2_storage/include/rosbag2_storage/storage_interfaces/base_write_interface.hpp

```diff
@@ -34,7 +35,7 @@ class ROSBAG2_STORAGE_PUBLIC BaseWriteInterface
 
   virtual void write(std::shared_ptr<const SerializedBagMessage> msg) = 0;
 
-  virtual void create_topic(const std::string & name, const std::string & type) = 0;
+  virtual void create_topic(const TopicWithType & topic) = 0;
```

Comment:
why do we need this change?

Reply:
It's not strictly speaking necessary at the moment, but since these two always come together, this better expresses the intent of what happens. When these types are used together with other information (e.g. rosbag info), this makes it easier to follow what's happening.